client/lib/invites/reducers/invites-create-validation.js

```diff
@@ -0,0 +1,27 @@
+/**
+ * External dependencies
+ */
+import { fromJS } from 'immutable';
+
+/**
+ * Internal dependencies
+ */
+import { action as ActionTypes } from 'lib/invites/constants';
+
+const initialState = fromJS( {
+	list: {},
+	errors: {}
+} );
+
+const reducer = ( state = initialState, payload ) => {
+	const { action } = payload;
+	switch ( action.type ) {
+		case ActionTypes.RECEIVE_CREATE_INVITE_VALIDATION_SUCCESS:
+			return state.setIn( [ 'list', action.siteId ], true );
+		case ActionTypes.RECEIVE_CREATE_INVITE_VALIDATION_ERROR:
+			return state.setIn( [ 'errors', action.siteId ], action.error );
+	}
+	return state;
+}
```

Comment:
I don't this matches with the API. The validation endpoint doesn't return errors for usernames or emails that don't validate. The endpoint returns an object that looks a bit like:

```
{
  errors: [ WP_Error objects ],
  success: [ user1, useremail@test.com ]
}
```